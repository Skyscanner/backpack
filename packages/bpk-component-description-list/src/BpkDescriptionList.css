@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-description-list{margin:0}.bpk-description-list__term{font-weight:700}.bpk-description-list__details{margin:0;margin-left:1.5rem}html[dir='rtl'] .bpk-description-list__details{margin-right:1.5rem;margin-left:0}
