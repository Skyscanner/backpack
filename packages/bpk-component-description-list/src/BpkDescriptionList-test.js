/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import BpkDescriptionList from './BpkDescriptionList';

describe('BpkDescriptionList', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <BpkDescriptionList>
          <dt>term</dt>
          <dd>description</dd>
        </BpkDescriptionList>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with a "className" attribute', () => {
    const tree = renderer
      .create(
        <BpkDescriptionList className="test-description-list">
          <dt>term</dt>
          <dd>description</dd>
        </BpkDescriptionList>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with other attributes', () => {
    const tree = renderer
      .create(
        <BpkDescriptionList title="this is a description list">
          <dt>term</dt>
          <dd>description</dd>
        </BpkDescriptionList>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
