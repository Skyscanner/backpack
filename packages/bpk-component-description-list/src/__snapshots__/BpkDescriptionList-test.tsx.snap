// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkDescriptionList should render correctly 1`] = `
<DocumentFragment>
  <dl
    class="bpk-description-list"
  >
    <dt>
      term
    </dt>
    <dd>
      description
    </dd>
  </dl>
</DocumentFragment>
`;

exports[`BpkDescriptionList should render correctly with a "className" attribute 1`] = `
<DocumentFragment>
  <dl
    class="bpk-description-list test-description-list"
  >
    <dt>
      term
    </dt>
    <dd>
      description
    </dd>
  </dl>
</DocumentFragment>
`;

exports[`BpkDescriptionList should render correctly with other attributes 1`] = `
<DocumentFragment>
  <dl
    class="bpk-description-list"
    title="this is a description list"
  >
    <dt>
      term
    </dt>
    <dd>
      description
    </dd>
  </dl>
</DocumentFragment>
`;
